File: VB/ChangeScrollValue/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraScheduler

Namespace ChangeScrollValue
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
			AddHandler schedulerControl1.MouseWheel, AddressOf schedulerControl1_MouseWheel
		End Sub

		Private Sub schedulerControl1_MouseWheel(ByVal sender As Object, ByVal e As MouseEventArgs)
			Dim topRowTime As TimeSpan = schedulerControl1.DayView.TopRowTime
			Dim ti As New TimeInterval(schedulerControl1.ActiveView.SelectedInterval.Start.Date, topRowTime)
			If (e.Delta > 0 AndAlso ti.Duration.TotalSeconds = 0) OrElse (e.Delta < 0 AndAlso ti.Duration.TotalHours = 24) Then
				Return
			End If
			Dim delta As TimeSpan = TimeSpan.FromMinutes(schedulerControl1.DayView.TimeScale.TotalMinutes * 2)
			If e.Delta > 0 Then
				If topRowTime.TotalMinutes < delta.TotalMinutes Then
					schedulerControl1.DayView.TopRowTime = delta
				Else
					schedulerControl1.DayView.TopRowTime = topRowTime.Subtract(delta)
				End If
			Else
				Dim day As TimeSpan = TimeSpan.FromDays(1)
				If (day.Subtract(topRowTime)).TotalMinutes < delta.TotalMinutes Then
					schedulerControl1.DayView.TopRowTime = day
				Else
					schedulerControl1.DayView.TopRowTime = topRowTime.Add(delta)
				End If
			End If
		End Sub
	End Class
End Namespace

File: CS/ChangeScrollValue/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraScheduler;

namespace ChangeScrollValue
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
            schedulerControl1.MouseWheel += new MouseEventHandler(schedulerControl1_MouseWheel);
        }

        void schedulerControl1_MouseWheel(object sender, MouseEventArgs e)
        {
            TimeSpan topRowTime = schedulerControl1.DayView.TopRowTime;
            TimeInterval ti = new TimeInterval(schedulerControl1.ActiveView.SelectedInterval.Start.Date, topRowTime);
            if ((e.Delta > 0 && ti.Duration.TotalSeconds == 0) || (e.Delta < 0 && ti.Duration.TotalHours == 24))
                return;
            TimeSpan delta = TimeSpan.FromMinutes(schedulerControl1.DayView.TimeScale.TotalMinutes * 2);
            if (e.Delta > 0)
            {
                if (topRowTime.TotalMinutes < delta.TotalMinutes)
                    schedulerControl1.DayView.TopRowTime = delta;
                else
                    schedulerControl1.DayView.TopRowTime = topRowTime - delta;
            }
            else { 
                TimeSpan day = TimeSpan.FromDays(1);
                if ((day - topRowTime).TotalMinutes < delta.TotalMinutes)
                    schedulerControl1.DayView.TopRowTime = day;
                else
                    schedulerControl1.DayView.TopRowTime = topRowTime + delta;
            }            
        }        
    }
}
